spring-cloud-kubernetes-controllers/spring-cloud-kubernetes-configuration-watcher/src/main/java/org/springframework/cloud/kubernetes/configuration/watcher/ConfigMapWatcherChangeDetector.java

```diff
@@ -38,60 +34,30 @@
  * @author Ryan Baxter
  * @author Kris Iyer
  */
-public abstract class ConfigMapWatcherChangeDetector extends KubernetesClientEventBasedConfigMapChangeDetector {
-
-	protected Log log = LogFactory.getLog(getClass());
+abstract class ConfigMapWatcherChangeDetector extends KubernetesClientEventBasedConfigMapChangeDetector
+		implements RefreshTrigger {
 
 	private final ScheduledExecutorService executorService;
 
-	protected ConfigurationWatcherConfigurationProperties k8SConfigurationProperties;
+	private final long refreshDelay;
 
-	public ConfigMapWatcherChangeDetector(CoreV1Api coreV1Api, ConfigurableEnvironment environment,
+	ConfigMapWatcherChangeDetector(CoreV1Api coreV1Api, ConfigurableEnvironment environment,
 			ConfigReloadProperties properties, ConfigurationUpdateStrategy strategy,
 			KubernetesClientConfigMapPropertySourceLocator propertySourceLocator,
 			KubernetesNamespaceProvider kubernetesNamespaceProvider,
 			ConfigurationWatcherConfigurationProperties k8SConfigurationProperties,
 			ThreadPoolTaskExecutor threadPoolTaskExecutor) {
 		super(coreV1Api, environment, properties, strategy, propertySourceLocator, kubernetesNamespaceProvider);
-
 		this.executorService = Executors.newScheduledThreadPool(k8SConfigurationProperties.getThreadPoolSize(),
 				threadPoolTaskExecutor);
-		this.k8SConfigurationProperties = k8SConfigurationProperties;
+		this.refreshDelay = k8SConfigurationProperties.getRefreshDelay().toMillis();
 	}
 
 	@Override
-	protected void onEvent(V1ConfigMap configMap) {
```

Comment:
this code was simplified and moved to `WatcherUtil::onEvent`. Logically, nothing has changed in that implementation